diff --git a/node_modules/@react-native-firebase/firestore/ios/RNFBFirestore/RNFBFirestoreCollectionModule.m b/node_modules/@react-native-firebase/firestore/ios/RNFBFirestore/RNFBFirestoreCollectionModule.m
index 588144f..2328966 100644
--- a/node_modules/@react-native-firebase/firestore/ios/RNFBFirestore/RNFBFirestoreCollectionModule.m
+++ b/node_modules/@react-native-firebase/firestore/ios/RNFBFirestore/RNFBFirestoreCollectionModule.m
@@ -259,10 +259,10 @@ RCT_EXPORT_METHOD(aggregateQuery
       NSString *reason = [@"Invalid Aggregate Type: " stringByAppendingString:aggregateType];
       [RNFBFirestoreCommon
           promiseRejectFirestoreException:reject
-                                    error:[NSException exceptionWithName:
-                                                           @"RNFB Firestore: Invalid Aggregate Type"
-                                                                  reason:reason
-                                                                userInfo:nil]];
+                                    error:[NSError errorWithDomain:@"RNFBFirestore"
+                          code:1
+                      userInfo:@{NSLocalizedDescriptionKey: reason}]
+      ];
       return;
     }
   }
